packages/opentelemetry-sdk-trace-base/src/export/BatchSpanProcessorBase.ts

```diff
@@ -166,6 +168,17 @@ export abstract class BatchSpanProcessorBase<T extends BufferConfig> implements
             }
           }
         );
+
+        const pendingResources = spans.map(span => span.resource)
+          .filter(resource => !resource.asyncAttributesHaveResolved());
+
+        // Avoid scheduling a promise to make the behavior more predictable and easier to test
+        if (pendingResources.length === 0) {
```

Comment:
Without this check, I had to make extensive changes to the tests because the fire-and-forget nature of L179